src/modules/iotjs_module_httpparser.c

```diff
@@ -213,7 +213,13 @@ static int iotjs_httpparserwrap_on_headers_complete(http_parser* parser) {
   iotjs_jargs_append_jval(&argv, &info);
 
   iotjs_jval_t res = iotjs_make_callback_with_result(&func, jobj, &argv);
-  bool ret = iotjs_jval_as_boolean(&res);
+
+  // if exception throw occurs in iotjs_make_callback_with_result, then the
+  // result can be an object.
+  int ret = 1;
+  if (iotjs_jval_is_boolean(&res) || iotjs_jval_is_object(&res)) {
```

Comment:
The error_flag is cleared before leaving `iotjs_make_callback_with_result` out. And, I want to fix this with minimal changes.